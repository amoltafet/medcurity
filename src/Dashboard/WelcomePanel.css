.WelcomePanelCard {
    font-family: Lato,sans-serif;
    color: white;
    background-color: #cc3333 !important;
    margin-top: .5vmax;
    margin-bottom: .5vmax;
    margin-left: 3vmax;
    margin-right: 1.5vmax;
    width: 55vmax;
    height: 23vmax;
}

.WelcomePanelImage {
    margin-left: 3vmax;
    margin-right: 1vmax;
    margin-top: 3.5vmax;
    margin-bottom: 1vmax;
    width: 12vmax;
    height: 12vmax; 
    box-shadow:  0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19)
}

.uvs-left {
    border-radius: 10px 10px 10px 10px;
    border: white;
    border-radius: .13vmax;
    box-shadow:  0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
}
    
    
.uvs-right {
    border-radius: 10px 10px 10px 10px;
}

.titleFont {
    font-size: 3.5vmax;
    margin-left: 1vmax;
    margin-right: 1vmax;
    margin-top: 2vmax;
    margin-bottom: 1vmax;
    font-family: Lato,sans-serif;
}
  
.cardSize {
    width: 90vmax;
    height: 15vmax;
}
  
  