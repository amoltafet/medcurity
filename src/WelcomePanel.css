.WelcomePanelCard {
    font-size: 5rem;
    text-align: left;
    color: white;
    margin-top: .5rem;
    margin-bottom: .5rem;
    margin-left: 3.5rem;
    margin-right: 1.5rem;
    background-color: #cc3333 !important;
    width: 60rem;
    height: 15rem;
}

.WelcomePanelImage{
    margin-left: 1rem;
    margin-top: 1rem;
    margin-bottom: 1rem;
    width: 10rem;
    height: 10rem;
}

.uvs-left {
    border-radius: 10px 10px 10px 10px;
}
    
    
.uvs-right {
    border-radius: 10px 10px 10px 10px;
}


  
  
  