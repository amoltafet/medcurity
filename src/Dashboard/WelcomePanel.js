import { Image, Card } from 'react-bootstrap'
import React from 'react';
import './WelcomePanel.css';
import './../Layout.css'
import 'bootstrap/dist/css/bootstrap.min.css'
import axios from "axios"
import { useEffect, useState } from "react";
// import env from "react-dotenv";

/**
* Creates and displays the main welcome panel for the user. 
* @return {WelcomePanel}
*/
const WelcomePanel = (props) => {
     axios.defaults.withCredentials = true;
    const [profilePic, setProfilePic] = useState("")
    useEffect(() => { if (props.user.userid)  axios.get(`${process.env.REACT_APP_BASE_URL}/api/getProfilePicture`, { params: { id: props.user.userid }} ).then((response) => { setProfilePic(response.data.profileImage) }); })

    

    return (
        <>
        <Card className="WelcomePanelCard uvs-left" style={{flexDirection: 'row'}}>
        <Image className="WelcomePanelImage uvs-left" variant="top" src={`data:image/png;base64,${profilePic}`} alt="" roundedCircle />
            <Card.Body className="cardSize">
               <Card.Title className="titleFont" >Welcome Back, {props.user.username} to the {props.pageTitle}!</Card.Title>
            </Card.Body> 
        </Card>
        </>
    );
}
export default WelcomePanel;