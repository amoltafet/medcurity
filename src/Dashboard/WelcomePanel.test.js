import renderer from 'react-test-renderer';
import WelcomePanel from './WelcomePanel';
import React from 'react';

test('renders WelcomePanel', () => {
  const tree = renderer.create(<WelcomePanel />).toJSON();
  expect(tree).toMatchSnapshot();
});