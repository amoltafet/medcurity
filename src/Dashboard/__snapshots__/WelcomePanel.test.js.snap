// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders WelcomePanel 1`] = `
<div
  className="WelcomePanelCard uvs-left card"
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <img
    alt=""
    className="WelcomePanelImage uvs-left rounded-circle"
    src="data:image/png;base64,"
    variant="top"
  />
  <div
    className="cardSize card-body"
  >
    <div
      className="titleFont card-title h5"
    >
      Welcome Back, 
      User
       to the 
      !
    </div>
  </div>
</div>
`;
